doc/contributing.rst

```diff
@@ -1,221 +1,577 @@
+
+Finally, add a few dependencies that are not needed for running MNE-Python but
+are needed for locally running our test suite or building our documentation:
+
+.. code-block:: console
+
+    $ pip install sphinx sphinx-gallery sphinx_bootstrap_theme sphinx_fontawesome memory_profiler
+    $ conda install sphinx-autobuild doc8  # linter for reStructuredText
+
+.. TODO: no longer needed? pip install sphinx-gallery should be fine...
+    $ cd $INSTALL_LOCATION
+    $ git clone git://github.com/sphinx-gallery/sphinx-gallery.git
+    $ cd sphinx-gallery
+    $ pip install -e .
+
+.. TODO: add this if sphinx-mermaid is officially adopted
+    Our documentation includes diagrams that are built automatically from text
+    using a tool called `mermaid`_. Originally a javascript tool, there is now
+    a command-line interface ``mermaid.cli`` as well as a plugin for ``sphinx``
+    (our documentation build tool) that allows mermaid blocks in the
+    documentation to be rendered into diagrams automatically.
+    .
+    $ wget -O - "https://dl.yarnpkg.com/debian/pubkey.gpg" | sudo apt-key add -
+    $ echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list > /dev/null
+    $ sudo apt update
+    $ sudo apt install yarn
+    Once yarn is installed, you can use it to install mermaid
+    $ cd $INSTALL_LOCATION
+    $ yarn add mermaid
+    $ yarn add mermaid.cli
+    $ pip install sphinxcontrib-mermaid
+
+.. note::
+
+    The commands ``pip install -e .`` and ``python setup.py develop`` both
+    install a python module into the current environment by creating a link to
+    the source code directory (instead of copying the code to pip's
+    ``site_packages`` directory, which is what normally happens. This means
+    that any edits you make to the MNE-Python source code will be reflected the
+    next time you open a Python interpreter and ``import mne`` (the ``-e`` flag
+    of ``pip`` stands for an "editable" installation).
+
+
+Basic git command reference
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Learning to work with git can take a long time, because it is a complex and
+powerful tool for managing versions of files across multiple users, each of
+whom have multiple copies of the codebase. We've already seen in the setup
+commands above a few of the basic git commands useful to an MNE-Python
+developer:
+
+.. code-block:: shell
+
+    # make a local copy of a repository:
+    git clone <URL_OF_REMOTE_REPO>
+
+    # connect a local copy to an additional remote:
+    git remote add <NICKNAME_OF_REMOTE> <URL_OF_REMOTE_REPO>
+
+    # get the current state of connected remote repos:
+    git fetch --all
+
+.. note::
+
+    You may have noticed that earlier we used ``git://`` instead of
+    ``https://`` in the address for the official "upstream" remote repository,
+    and in the addresses for mayavi and sphinx-gallery.
+    ``git://`` addresses are read-only, which means you can *pull* the official
```

Comment:
hmmm? I thought it meant that it is to be accessed using SSH protocol, no?

Reply:
`git@` addresses use SSH. `git://` addresses are read-only